src/core/utils/modal.tsx

```diff
@@ -65,6 +65,7 @@ function Modal({
           classNames
         )}
         role="dialog"
+        aria-label={screenReaderModalDescriptionText}
```

Comment:
Please reconsider this approach. 

As I see it this will duplicate content that is already target accessibility tools in `modal__screen-reader-description`. Have you looked into other approaches e.g.:

- Dropping the `modal__screen-reader-description` element in favour of this label
- Dropping `aria-label` in favourt of [`aria-describedby`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-describedby) which could point to the `modal__screen-reader-description`/`modal-${modalId}` element